tls: Control enforce_rsa_key_usage (#24923)
envoyproxy/envoy: docs/root/_include/ssl_stats.rst

   curves.<curve>, Counter, Total successful TLS connections that used ECDHE curve <curve>
   sigalgs.<sigalg>, Counter, Total successful TLS connections that used signature algorithm <sigalg>
   versions.<version>, Counter, Total successful TLS connections that used protocol version <version>

>    was_key_usage_invalid, Counter, Total successful TLS connections that used protocol version has invalid RSA key usage

Reviewer: ```suggestion
   was_key_usage_invalid, Counter, Total successful TLS connections that used an invalid RSA key usage
```

Can you link to some docs (either other envoy docs, or boringssl docs) that describe what this means?

>    was_key_usage_invalid, Counter, Total successful TLS connections that used an `invalid keyUsage extension <https://github.com/google/boringssl/blob/6f13380d27835e70ec7caf807da7a1f239b10da6/ssl/internal.h#L3117>`_. (This is not avaiable in BoringSSL FIPS yet due to `issue <https://github.com/envoyproxy/envoy/issues/28246>`_)

Reviewer 2: nit: s/issue <...>/issue #28246<...>/
